src/Graphics/Texture2D.cs

```diff
@@ -158,7 +158,7 @@ int elementCount
 				w = Math.Max(Width >> level, 1);
 				h = Math.Max(Height >> level, 1);
 			}
-
+			var elementSize = Marshal.SizeOf(typeof(T));
```

Comment:
Only one `var`, too! Aside from this the patch is good, so just rinse and repeat for Get and 3D/Cube.